ibm/resource_ibm_kms_key_policy.go

```diff
@@ -0,0 +1,476 @@
+							Description:  "Specifies the key rotation time interval in months",
+						},
+					},
+				},
+			},
+			"dual_auth_delete": {
+				Type:         schema.TypeList,
+				Optional:     true,
+				Computed:     true,
+				AtLeastOneOf: []string{"rotation", "dual_auth_delete"},
+				Description:  "Data associated with the dual authorization delete policy.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The v4 UUID used to uniquely identify the policy resource, as specified by RFC 4122.",
+						},
+						"crn": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Cloud Resource Name (CRN) that uniquely identifies your cloud resources.",
+						},
+						"created_by": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier for the resource that created the policy.",
+						},
+						"creation_date": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The date the policy was created. The date format follows RFC 3339.",
+						},
+						"updated_by": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier for the resource that updated the policy.",
+						},
+						"last_update_date": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Updates when the policy is replaced or modified. The date format follows RFC 3339.",
+						},
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "If set to true, Key Protect enables a dual authorization policy on a single key.",
+						},
+					},
+				},
+			},
+			ResourceName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the resource",
+			},
+			ResourceCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+			ResourceStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the resource",
+			},
+		},
+	}
+}
+func resourceIBMKmsKeyPolicyCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	kpAPI, err := meta.(ClientSession).keyManagementAPI()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	rContollerClient, err := meta.(ClientSession).ResourceControllerAPIV2()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Println("Entering the Policies++++++++>>>>>>>>")
+
+	instanceID := d.Get("instance_id").(string)
+	endpointType := d.Get("endpoint_type").(string)
+	key_id := d.Get("key_id").(string)
+
+	rContollerApi := rContollerClient.ResourceServiceInstanceV2()
+
+	instanceData, err := rContollerApi.GetInstance(instanceID)
+	if err != nil {
+		return diag.FromErr(err)
```

Comment:
return diag.FromErr(err) == > return diag.FromErr(fmt.Errof("[ERROR] Error getting KMS Instance during creation of policy %s", err))